sound/soc/sof/intel/hda-stream.c

```diff
@@ -224,13 +226,18 @@ int hda_dsp_stream_put(struct snd_sof_dev *sdev, int direction, int stream_tag)
 			s->opened = false;
 			found = true;
 		} else if (s->direction == SNDRV_PCM_STREAM_CAPTURE) {
-			active_capture_stream = true;
+			struct hdac_ext_stream *stream = stream_to_hdac_ext_stream(s);
+			struct sof_intel_hda_stream *hda_stream;
+
+			hda_stream = container_of(stream, struct sof_intel_hda_stream, hda_stream);
+			if (!hda_stream->d0i3_compatible)
+				active_capture_stream = true;
 		}
 	}
 
 	spin_unlock_irq(&bus->reg_lock);
 
-	/* Enable DMI L1 entry if there are no capture streams open */
+	/* Enter DMI L1 if there are no non-d0i3-compatible capture streams open */
```

Comment:
"no non-d0i3-compatile" is bit of a mouthful, but I'm ok if you are :) 

If you make an additional spin, my proposal:
"Allow DMI L1 if all remaining streams support it */

Reply:
ack. done